dev.c

```diff
@@ -607,52 +642,88 @@ static int copy_in_read_data_iovec(struct iov_iter *iter,
 	return 0;
 }
 
-static int fuse_notify_read_data(struct fuse_conn *conn, unsigned int size,
-				struct iov_iter *iter)
+static int __fuse_notify_read_data_slowpath(struct fuse_conn *conn,
```

Comment:
Lot of duplication between the functions below. Can you instead pass in fastpath as argument to a common fn or reorg however you prefer so common code gets moved out?